### ISSUE TITLE
useTreeData append stops working after selecting a key using setSelectedKeys

### ISSUE THREAD

**Issue by nkalpak** [adobe/react-spectrum#3288]
# 🐛 Bug Report 

`useTreeData` append stops working after selecting a key using `setSelectedKeys`

## 🤔 Expected Behavior

`append` should work at all times

## 😯 Current Behavior

`append` quits working after an item is selected using `setSelectedKeys`

## 💁 Possible Solution

/

## 🔦 Context

I am trying to build a custom tree view component using the `useTreeData` hook

## 💻 Code Sample

https://codesandbox.io/s/currying-https-vyq7ri?file=/src/App.tsx:363-364

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| @react-stately/data   | 3.5.1
| Browser          | Google Chrome
| Operating System | Windows 10

**Comment by snowystinger** [comment]
Add Item fails the second time you call it every time, regardless of the call to setSelectedKeys. Might make sure you aren't creating two items with the same id?

**Comment by nkalpak** [comment]
Yeah, if you "add item" twice in succession in the codesandbox, I don't expect to see 2 items because the item with that id is already added.

This issue is about the item not being added _at all_ if you _first change the selection_ in the tree data state. The following video might help to illustrate the flow:

https://user-images.githubusercontent.com/57584132/178211393-fca43a33-6d7d-4464-8d48-7788925443b3.mp4

Basically, there are two flows:
1. Click on "add item" once
2. All good, item with ID `8324823` is added

Alternatively:
1. Click on "select 1"
2. This calls `setSelectedKeys` to update the selection with the key `1`
3. Click on "add item" once
4. Item with ID `8324823` isn't added, even though it doesn't already exist in the state

**Comment by snowystinger** [comment]
Took another look, appears to be related to StrictMode. If you remove StrictMode from the codesandbox, it works as expected. StrictMode is being addressed under the umbrella Issue https://github.com/adobe/react-spectrum/issues/779
